plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/properties/EditUntypedSubappInterfaceDataSection.java

```diff
@@ -194,4 +203,24 @@ private static boolean isConnected(final VarDeclaration varDeclaration) {
 			return !varDeclaration.getInputConnections().isEmpty() || !varDeclaration.getOutputConnections().isEmpty();
 		}
 	}
+
+	private static class CopyVarDataFromTableAction extends CopyDataFromTableAction {
+		@Override
+		public void run(final NatTable natTable, final KeyEvent event) {
+			final ListDataProvider<?> provb = (ListDataProvider<?>) NatTableWidgetFactory.getDataLayer(natTable)
+					.getDataProvider();
+			final SelectionLayer selectionLayer = NatTableWidgetFactory.getSelectionLayer(natTable);
+			if (selectionLayer != null && selectionLayer.getSelectedCells().size() == 1) {
+				final var cell = selectionLayer.getSelectedCells().iterator().next();
+				final var rowObj = provb.getRowObject(cell.getRowIndex());
+				if (cell.getColumnIndex() == 1 && rowObj instanceof final VarDeclaration decl) {
+					final Clipboard clipboard = new Clipboard(Display.getDefault());
+					clipboard.setContents(new String[] { PackageNameHelper.getFullTypeName(decl.getType()) },
+							new Transfer[] { TextTransfer.getInstance() });
```

Comment:
I believe this will simply copy the full qualified name instead of adding an import when pasted into a type cell? I would suggest creating a separate transfer that also includes the required imports and then add them in the destination type if possible and necessary.